File: 2026/TPs/TP01/Inverter.java
import java.util.Scanner;

class Inverter{
    public static boolean isFim(String str){
        return str.length() == 3 &&
               str.charAt(0) == 'F' &&
               str.charAt(1) == 'I' &&
               str.charAt(2) == 'M';
    }

    public static String inverter(String str){
        String resp = "";

        for(int i = str.length() - 1; i >= 0; i--){
            resp += str.charAt(i);
        }

        return resp;
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        String linha = sc.nextLine();

        while(true){
            if(isFim(linha)){
                break;
            }

            System.out.println(inverter(linha));

            linha = sc.nextLine();
        }

        sc.close();
    }
}

File: 2026/TPs/TP01/Inverter.c
#include <stdio.h>

int tamanho(char str[]){
    int i = 0;
    while(str[i] != '\0'){
        i++;
    }
    return i;
}

int isFim(char str[]){
    return tamanho(str) == 3 &&
           str[0] == 'F' &&
           str[1] == 'I' &&
           str[2] == 'M';
}

void inverter(char str[]){
    int tam = tamanho(str);

    for(int i = tam - 1; i >= 0; i--){
        printf("%c", str[i]);
    }
    printf("\n");
}

int main(){
    char linha[1000];

    fgets(linha, 1000, stdin);

    while(1){
        int i = 0;
        while(linha[i] != '\0'){
            if(linha[i] == '\n'){
                linha[i] = '\0';
                break;
            }
            i++;
        }

        if(isFim(linha)){
            break;
        }

        inverter(linha);

        fgets(linha, 1000, stdin);
    }

    return 0;
}